lib/modules/manager/gradle/artifacts.ts

```diff
@@ -148,13 +154,26 @@ export async function updateArtifacts({
       .map(quote)
       .join(' ')}`;
 
+    if (verificationMetadataFile) {
+      const hashTypes: string[] = [];
+      const verificationMetadata = await readLocalFile(
+        verificationMetadataFile
+      );
+      ['md5', 'sha1', 'sha256', 'sha512', 'pgp'].forEach((hashType) => {
+        if (verificationMetadata?.includes(`<${hashType}`)) {
+          hashTypes.push(hashType);
+        }
+      });
+      cmd += ` --write-verification-metadata ${hashTypes.join(',')}`;
```

Comment:
Quote from [Gradle Docs, Bootstrapping dependency verification](https://docs.gradle.org/current/userguide/dependency_verification.html#sec:bootstrapping-verification):
 
> Bootstrapping can either be used to create a file from the beginning, or also to _update_ an existing file with new information. Therefore, it’s recommended to always use the same parameters once you started bootstrapping.
> 
> The dependency verification file can be generated with the following CLI instructions:
> 
> `gradle --write-verification-metadata sha256 help`

How do we make sure we're always using the same command? It sounds like mixing different commands will cause problems.

Also it looks like Gradle recommends starting with the `help` command first.

> There are dependencies that Gradle _cannot_ discover this way. In particular, you will notice that the CLI above uses the `help` task. If you don’t specify any task, Gradle will automatically run the default task and generate a configuration file at the end of the build too.
>
> The difference is that Gradle _may_ discover more dependencies and artifacts depending on the tasks you execute. As a matter of fact, Gradle cannot automatically discover _detached configurations_,  which are basically dependency graphs resolved as an internal  implementation detail of the execution of a task: they are not, in  particular, declared as an input of the task because they effectively  depend on the configuration of the task at execution time.
>
> A good way to start is just to use the simplest task, `help`,  which will discover as much as possible, and if subsequent builds fail  with a verification error, you can re-execute generation with the  appropriate tasks to "discover" more dependencies.
>
> Gradle won’t verify either checksums or signatures of plugins which use their own HTTP clients. Only plugins which use the infrastructure provided by Gradle for performing requests will see their requests verified.

Reply:
I thought the `parameters` in the docs referred to the hash type, and that it would be fine to use the `dependencies` task that is being run anyway.

But your comment got me thinking that it would be safest to just have an extra exec call just for the verification metadata, separated from the lockfile command. That way we can be sure that the same task (`help`) is used every time.

Reply:
sounds good